### ISSUE TITLE
[BUG] _processErrorPageUrl($url) is broken when Typo returns NullSite

### ISSUE THREAD

**Issue by theonlymb** [aaw-team/pagenotfoundhandling#9]
Hey! First of all - sorry if I missed any conventions opening issues into your Repo.

Our log (in Typo3 V9.5) was continuous showing erros for URLs like _www.ourdomain.com/22.html?L=11088.html&cHash=67c321f1d89a4e8c3622cb0fceacda59&tx_ttnews%%5Btt_news%%5D=123_. 

The error is: **Exception handler (WEB): Uncaught TYPO3 Exception: Call to undefined method TYPO3\CMS\Core\Site\Entity\NullSite::getRouter()**.

It seems like some old cHashes are called - from whatever client in the web. The interesting thing is, you can produce this error with calling a non-existing page id, e.g. _www.ourdomain.com/index.php&?id=1234_. After having a look at the TYPO3.CMS repo and your code, I think I could figure out the error in your _PagenotfoundController.php_:

```
// Finally create the new $url
$url = (string)$site->getRouter()->generateUri(
     $errorPageUid,
     $queryParams
);
```
In the above described cases typo3 seems to return a NullSite (see https://github.com/TYPO3/TYPO3.CMS/blob/master/typo3/sysext/core/Classes/Site/Entity/NullSite.php) which isn't having a method _getRouter()_ to generate an URI.

I will try to fix this for myself by checking if _$site_ is an instance of _NullSite_ and then returning an hard-coded error URI for our page. It would be cool if you could fit the code in the right way to get a language-specific and correct error URL if a NullSite is returned.

Thanks in advance, Regards
Markus

**Comment by christianfutterlieb** [comment]
Hi @mb-karo 

Thanks for your report, I see the problem. It seems we rely on `$request->getAttribute('site', null)` returning a `\TYPO3\CMS\Core\Site\Entity\Site` without checking. So, we could receive anything without noticing.

As noted above: the `\TYPO3\CMS\Core\Site\Entity\SiteInterface` does not define the `getRouter()` method, it exists only in `\TYPO3\CMS\Core\Site\Entity\Site`.. Looks like we're stuck with either checking for that exact type, or testing for the interface and method existence.

Tomorrow I'll push a fix for master and 2_5.

**Comment by christianfutterlieb** [comment]
Hi @mb-karo

I just pushed a fix for 2_5 and master. Can you please quick test if your problem is resolved (and you're not getting any other problems)? When everything works fine, I'll release the fix shortly.

**Comment by theonlymb** [comment]
Hey Christian,

it works! Thanks!!

**Comment by christianfutterlieb** [comment]
Thanks for the test, I just released version 2.5.3.
